ksqldb-execution/src/main/java/io/confluent/ksql/execution/plan/ForeignKeyTableTableJoin.java

```diff
@@ -42,24 +45,37 @@ public ForeignKeyTableTableJoin(
       final ExecutionStepPropertiesV1 props,
       @JsonProperty(value = "joinType", required = true)
       final JoinType joinType,
-      @JsonProperty(value = "leftJoinColumnName", required = true)
-      final ColumnName leftJoinColumnName,
+      @JsonProperty(value = "leftJoinColumnName")
+      final Optional<ColumnName> leftJoinColumnName,
       @JsonProperty(value = "formats", required = true)
       final Formats formats,
       @JsonProperty(value = "leftSource", required = true)
       final ExecutionStep<KTableHolder<KLeftT>> leftSource,
       @JsonProperty(value = "rightSource", required = true)
-      final ExecutionStep<KTableHolder<KRightT>> rightSource
+      final ExecutionStep<KTableHolder<KRightT>> rightSource,
+      @JsonProperty(value = "leftJoinExpression")
+      final Optional<Expression> leftJoinExpression
   ) {
     this.properties = requireNonNull(props, "props");
     this.joinType = requireNonNull(joinType, "joinType");
     if (joinType == JoinType.OUTER) {
       throw new IllegalArgumentException("OUTER join not supported.");
     }
-    this.leftJoinColumnName = requireNonNull(leftJoinColumnName, "leftJoinColumnName");
```

Comment:
We can still require non-null for `Optional` types. It'd be nice to add this back, and also add it for `leftJoinExpression` below.